project/StdBits.scala

```diff
@@ -233,6 +233,86 @@ object StdBits {
     )
   }
 
+  /** Extracts native libraries from `std-microsoft`.
+    * In particular from JNA.
+    *
+    * The list of the native libraries is listed in
+    * <a href="https://github.com/enso-org/enso/blob/7e0c6373b55bdf976562bce899f2fe6af7c258c0/test/Base_Tests/data/native_libs.json#L2-L25">
+    *   test/Base_Tests/data/native_libs.json
+    * </a>
+    *
+    * @param jnaJar Path to `jna-wrapper/assembly`.
+    */
+  def extractNativeLibsFromMicrosoft(
+    microsoftPolyglotRoot: File,
+    microsoftNativeLibs: File,
+    jnaJar: File,
+    logger: ManagedLogger,
+    moduleName: String,
+    cacheStoreFactory: CacheStoreFactory,
+    previousRun: Option[AnalysisOfExtractedNativeLibs]
+  ): AnalysisOfExtractedNativeLibs = {
+    if (previousRun.exists(!_.isOutdated)) {
+      return previousRun.get
+    }
+    val nativeCodeEntries =
+      JARUtils.readNativeCodeEntriesFromManifest(jnaJar.toPath)
+    val (expectedOsName, targetOs) = if (Platform.isWindows) {
```

Comment:
I feel like I've already seen code that handles OS and architecture naming. Let's keep it in one place, please.